### PR TITLE
Newlib updates

### PR DESCRIPTION
**Pull request by jasmin-j**
Adding missing newlib configure options.
Based on @bhundven's branch for newlib snapshot releases.

This closes #320

BR Jasmin

### CODE REVIEW

**Comment by bhundven** [comment]
I merged #320, Please rebase your commits.

**Comment by jasmin-j** [comment]
> I merged #320, Please rebase your commits.

Done!

File: config/libc/newlib.in.2 [hunk 1]
```diff
@@ -65,6 +65,77 @@ config LIBC_NEWLIB_NANO_FORMATTED_IO
       with size constraint issues.  This option does not affect wide-char
       formatted I/O functions.
 
+config LIBC_NEWLIB_REGISTER_FINI
+    bool
+    prompt "Enable Register finalization"
+    depends on LIBC_NEWLIB_2_2_or_later
+    help
+      Enable finalization function registration using atexit.
```
Comment (stilor) [inline config/libc/newlib.in.2]: So, is this "finalizing registers" or "registering a finalization function"? Please correct either the prompt, or the help message.

File: config/libc/newlib.in.2 [hunk 2]
```diff
@@ -65,6 +65,77 @@ config LIBC_NEWLIB_NANO_FORMATTED_IO
...
+config LIBC_NEWLIB_REENT_SMALL
+    bool
+    prompt "Enable Small reent structure"
+    depends on LIBC_NEWLIB_2_2_or_later
+    help
+      Enable small reentrant struct support.
+
+config LIBC_NEWLIB_ATEXIT_DYNAMIC_ALLOC
+    bool
+    prompt "Disable atexit dynamic allocation"
```
Comment (stilor) [inline config/libc/newlib.in.2]: The name of the option and the prompt are counter-intuitively named. One would expect the check for `"${CT_LIBC_NEWLIB_ATEXIT_DYNAMIC_ALLOC} = y` to verify that the "atexit dynamic allocation is enabled".

There are similar misnomers below.

I think the best approach to keep the option names, add `default y`, and correct prompts/help messages; and invert the checks in newlib.sh.

Alternatively, the options may be renamed to LIBC_NEWLIB_NO_ATEXIT_DYNAMIC_ALLOC, etc.

File: config/libc/newlib.in.2 [hunk 3]
```diff
@@ -65,6 +65,77 @@ config LIBC_NEWLIB_NANO_FORMATTED_IO
...
+    depends on LIBC_NEWLIB_2_2_or_later
+    help
+      Disable dynamic allocation of atexit entries.
+      Most hosts and targets have it enabled in configure.host.
+
+config LIBC_NEWLIB_FVWRITE_IN_STREAMIO
+    bool
+    prompt "Disable vector buffer in stream IO"
+    depends on LIBC_NEWLIB_2_2_or_later
```
Comment (stilor) [inline config/libc/newlib.in.2]: Another option where "yes" means "disable".

Please check the rest.

**Comment by stilor** [comment]
Superseded by https://github.com/crosstool-ng/crosstool-ng/pull/515

**Comment by stilor** [comment]
Sorry, should've been https://github.com/crosstool-ng/crosstool-ng/pull/519

**Pull request closed without merging**